core/src/main/scala/com/fortysevendeg/mvessel/Connection.scala

```diff
@@ -4,31 +4,37 @@ import java.sql.{Connection => SQLConnection, DatabaseMetaData => SQLDatabaseMet
 import java.util.Properties
 import java.util.concurrent.Executor
 
-import com.fortysevendeg.mvessel.logging.{AndroidLogWrapper, LogWrapper}
+import com.fortysevendeg.mvessel.api.DatabaseProxyFactory
+import com.fortysevendeg.mvessel.logging.LogWrapper
 import com.fortysevendeg.mvessel.metadata.DatabaseMetaData
 import com.fortysevendeg.mvessel.statement.{PreparedStatement, Statement}
-import com.fortysevendeg.mvessel.util.DatabaseUtils._
+import com.fortysevendeg.mvessel.util.DatabaseUtils.WrapSQLException
 
 class Connection(
+  databaseWrapperFactory: DatabaseProxyFactory,
   val databaseName: String,
   val timeout: Long = 0,
   val retryInterval: Int = 50,
   val flags: Int = 0,
-  val logWrapper: LogWrapper = new AndroidLogWrapper())
+  val logWrapper: LogWrapper)
   extends SQLConnection
   with WrapperNotSupported {
 
-  protected def createDatabase(): Option[Database] =
-    Some(new Database(databaseName, timeout, retryInterval, flags))
-
   protected def defaultAutoCommit(): Boolean = false
 
   private[this] def createPreparedStatement(
     sql: String,
     columnName: Option[String] = None
     ) = columnName match {
-    case Some(c) => new PreparedStatement(sql, this, Some(c))
-    case _ => new PreparedStatement(sql, this)
+    case Some(c) => new PreparedStatement(
+      sql = sql,
+      connection = this,
+      columnGenerated = Some(c),
+      logWrapper = logWrapper)
+    case _ => new PreparedStatement(
+      sql = sql,
+      connection = this,
+      logWrapper = logWrapper)
```

Comment:
Should this block equivalent to
new PreparedStatement(
  			sql = sql,
  			connection = this,
			columnGenerated = c,
  			logWrapper = logWrapper)

?